src/app/pages/quickMaps/pages/projection/projectionFoodSources/projectionFoodSources.component.ts

```diff
@@ -37,52 +93,294 @@ export class ProjectionFoodSourcesComponent implements AfterViewInit {
+
+          this.subscriptions.push(
+            this.quickMapsService.parameterChangedObs.subscribe(() => {
+              this.micronutrientName = this.quickMapsService.micronutrientIdName;
+              this.title = 'Projection Food Sources for ' + this.micronutrientName;
+              this.card.title = this.title;
+              this.init(
+                this.currentDataService.getProjectedFoodSourceData(
+                  this.quickMapsService.countryId,
+                  this.quickMapsService.micronutrientId,
+                  this.currentImpactScenario.id,
+                ),
+              );
+            }),
+          );
+        } else if (null != this.dialogData) {
+          // if displayed within a dialog use the data passed in
+          // this.init(Promise.resolve(this.dialogData.dataIn.data));
+          // this.tabGroup.selectedIndex = this.dialogData.dataIn.selectedTab;
+          // this.cdr.detectChanges();
+        }
+      });
+  }
 
-      // respond to parameter updates
-      this.subscriptions.push(
-        this.quickMapsService.parameterChangedObs.subscribe(() => {
-          // this.init(this.currentDataService.getMonthlyFoodGroups(
-          //   this.quickMapsService.countryId,
-          //   [this.quickMapsService.micronutrientId],
-          //   this.quickMapsService.popGroupId,
-          //   this.quickMapsService.mndDataId,
-          // ));
-        })
+  private init(dataPromise: Promise<Array<ProjectedFoodSourcesData>>): void {
+    this.loadingSrc.next(true);
+    dataPromise
+      .then((data: Array<ProjectedFoodSourcesData>) => {
+        this.data = data;
+        // console.log('projFoodSource data: ', data);
+        if (null == data) {
+          throw new Error('data error');
+        }
+
+        // Select current countries
+        const filteredByCountry: Array<ProjectedFoodSourcesData> = data.filter(
+          (item: ProjectedFoodSourcesData) => item.country === this.selectedCountry,
+        );
+
+        // Filter by current impact scenario
+        const filteredByScenario: Array<ProjectedFoodSourcesData> = filteredByCountry.filter(
+          (item: ProjectedFoodSourcesData) => item.scenario === this.currentImpactScenario.name,
+        );
+
+        // Filter by year and populate the array of objects
+        const filteredData = new Array<ProjectedFoodSourcesPeriod>();
+        const filteredTableDataArray = [];
+
+        const quinquennialPeriod = [...new Set(data.map((item) => item.year))];
+        quinquennialPeriod.forEach((currentYear) => {
+          const filteredTableData = new Array<ProjectedFoodSourcesTable>();
+
+          // Populate data for chart.js
+          filteredData.push({
+            year: currentYear,
+            data: {
+              cassava: filteredByScenario
+                .filter((item) => item.year === currentYear)
+                .filter((item) => item.commodity === 'Cassava')
```

Comment:
It would be nice if the data here was in a state where we didn't need hard-coded strings here and we could loop like you have for "filteredTableData".

Reply:
done